components/prism-hoon.js

```diff
@@ -0,0 +1,42 @@
+Prism.languages.hoon = {
+  constant: [
+    {
+      pattern: /(%\.n)|(%\.y)|(%[\w-]*[\w\d]?)/,
+    }
+  ],
+  comment: {
+    pattern: /::.*/,
+    lookahead: true,
+    greedy: true
+  },
+  "function-name": [
+    {
+      pattern: /(\+[-+]  )?([a-z]([a-z0-9-]*[a-z0-9])?)/
+    },
+  ],
+  "class-name": [
+    {
+      pattern: /@([a-z])?([A-Za-z0-9-]*[A-Za-z0-9])?/,
+      lookahead: true,
+      greedy: true
+    },
+    {
+      pattern: /\*/
+    }
```

Comment:
Instead of `{ pattern: /regex/ }` you can just use `/regex/`. Please use the simple regex notation wherever possible.